<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>978</width>
    <height>662</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SystemC Mips Simulator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="media.qrc">
    <normaloff>:/icons/Latest_Article.png</normaloff>:/icons/Latest_Article.png</iconset>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0">
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Instruction Memory</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QTableWidget" name="tableInstructionMemory">
         <column>
          <property name="text">
           <string>Address</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Code</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Instruction</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Register File</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QTableWidget" name="tableRegFile">
           <property name="font">
            <font>
             <pointsize>7</pointsize>
            </font>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QAbstractScrollArea::AdjustToContents</enum>
           </property>
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>60</number>
           </attribute>
           <attribute name="horizontalHeaderMinimumSectionSize">
            <number>70</number>
           </attribute>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="verticalHeaderDefaultSectionSize">
            <number>15</number>
           </attribute>
           <row>
            <property name="text">
             <string>0</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>1</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>2</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>3</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>4</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>5</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>6</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>7</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>8</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>9</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>10</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>11</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>12</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>13</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>14</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>15</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>16</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>17</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>18</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>19</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>20</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>21</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>22</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>23</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>24</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>25</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>26</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>27</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>28</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>29</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>30</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>31</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>32</string>
            </property>
           </row>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Address</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
           <item row="0" column="0">
            <property name="text">
             <string>$zero</string>
            </property>
           </item>
           <item row="0" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="0" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="1" column="0">
            <property name="text">
             <string>$at</string>
            </property>
           </item>
           <item row="1" column="1">
            <property name="text">
             <string>1</string>
            </property>
           </item>
           <item row="1" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="2" column="0">
            <property name="text">
             <string>$v0</string>
            </property>
           </item>
           <item row="2" column="1">
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item row="2" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="3" column="0">
            <property name="text">
             <string>$v1</string>
            </property>
           </item>
           <item row="3" column="1">
            <property name="text">
             <string>3</string>
            </property>
           </item>
           <item row="3" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="4" column="0">
            <property name="text">
             <string>$a0</string>
            </property>
           </item>
           <item row="4" column="1">
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item row="4" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="5" column="0">
            <property name="text">
             <string>$a1</string>
            </property>
           </item>
           <item row="5" column="1">
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item row="5" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="6" column="0">
            <property name="text">
             <string>$s2</string>
            </property>
           </item>
           <item row="6" column="1">
            <property name="text">
             <string>6</string>
            </property>
           </item>
           <item row="6" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="7" column="0">
            <property name="text">
             <string>$a3</string>
            </property>
           </item>
           <item row="7" column="1">
            <property name="text">
             <string>7</string>
            </property>
           </item>
           <item row="7" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="8" column="0">
            <property name="text">
             <string>$t0</string>
            </property>
           </item>
           <item row="8" column="1">
            <property name="text">
             <string>8</string>
            </property>
           </item>
           <item row="8" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="9" column="0">
            <property name="text">
             <string>$t1</string>
            </property>
           </item>
           <item row="9" column="1">
            <property name="text">
             <string>9</string>
            </property>
           </item>
           <item row="9" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="10" column="0">
            <property name="text">
             <string>$t2</string>
            </property>
           </item>
           <item row="10" column="1">
            <property name="text">
             <string>10</string>
            </property>
           </item>
           <item row="10" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="11" column="0">
            <property name="text">
             <string>$t3</string>
            </property>
           </item>
           <item row="11" column="1">
            <property name="text">
             <string>11</string>
            </property>
           </item>
           <item row="11" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="12" column="0">
            <property name="text">
             <string>$t4</string>
            </property>
           </item>
           <item row="12" column="1">
            <property name="text">
             <string>12</string>
            </property>
           </item>
           <item row="12" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="13" column="0">
            <property name="text">
             <string>$t5</string>
            </property>
           </item>
           <item row="13" column="1">
            <property name="text">
             <string>13</string>
            </property>
           </item>
           <item row="13" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="14" column="0">
            <property name="text">
             <string>$t6</string>
            </property>
           </item>
           <item row="14" column="1">
            <property name="text">
             <string>14</string>
            </property>
           </item>
           <item row="14" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="15" column="0">
            <property name="text">
             <string>$t7</string>
            </property>
           </item>
           <item row="15" column="1">
            <property name="text">
             <string>15</string>
            </property>
           </item>
           <item row="15" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="16" column="0">
            <property name="text">
             <string>$s0</string>
            </property>
           </item>
           <item row="16" column="1">
            <property name="text">
             <string>16</string>
            </property>
           </item>
           <item row="16" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="17" column="0">
            <property name="text">
             <string>$s1</string>
            </property>
           </item>
           <item row="17" column="1">
            <property name="text">
             <string>17</string>
            </property>
           </item>
           <item row="17" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="18" column="0">
            <property name="text">
             <string>$s2</string>
            </property>
           </item>
           <item row="18" column="1">
            <property name="text">
             <string>18</string>
            </property>
           </item>
           <item row="18" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="19" column="0">
            <property name="text">
             <string>$s3</string>
            </property>
           </item>
           <item row="19" column="1">
            <property name="text">
             <string>19</string>
            </property>
           </item>
           <item row="19" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="20" column="0">
            <property name="text">
             <string>$s4</string>
            </property>
           </item>
           <item row="20" column="1">
            <property name="text">
             <string>20</string>
            </property>
           </item>
           <item row="20" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="21" column="0">
            <property name="text">
             <string>$s5</string>
            </property>
           </item>
           <item row="21" column="1">
            <property name="text">
             <string>21</string>
            </property>
           </item>
           <item row="21" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="22" column="0">
            <property name="text">
             <string>$s6</string>
            </property>
           </item>
           <item row="22" column="1">
            <property name="text">
             <string>22</string>
            </property>
           </item>
           <item row="22" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="23" column="0">
            <property name="text">
             <string>$s7</string>
            </property>
           </item>
           <item row="23" column="1">
            <property name="text">
             <string>23</string>
            </property>
           </item>
           <item row="23" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="24" column="0">
            <property name="text">
             <string>$t8</string>
            </property>
           </item>
           <item row="24" column="1">
            <property name="text">
             <string>24</string>
            </property>
           </item>
           <item row="24" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="25" column="0">
            <property name="text">
             <string>$t9</string>
            </property>
           </item>
           <item row="25" column="1">
            <property name="text">
             <string>25</string>
            </property>
           </item>
           <item row="25" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="26" column="0">
            <property name="text">
             <string>$k0</string>
            </property>
           </item>
           <item row="26" column="1">
            <property name="text">
             <string>26</string>
            </property>
           </item>
           <item row="26" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="27" column="0">
            <property name="text">
             <string>$k1</string>
            </property>
           </item>
           <item row="27" column="1">
            <property name="text">
             <string>27</string>
            </property>
           </item>
           <item row="27" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="28" column="0">
            <property name="text">
             <string>$gp</string>
            </property>
           </item>
           <item row="28" column="1">
            <property name="text">
             <string>28</string>
            </property>
           </item>
           <item row="28" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="29" column="0">
            <property name="text">
             <string>$sp</string>
            </property>
           </item>
           <item row="29" column="1">
            <property name="text">
             <string>29</string>
            </property>
           </item>
           <item row="29" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="30" column="0">
            <property name="text">
             <string>$fp</string>
            </property>
           </item>
           <item row="30" column="1">
            <property name="text">
             <string>30</string>
            </property>
           </item>
           <item row="30" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="31" column="0">
            <property name="text">
             <string>$ra</string>
            </property>
           </item>
           <item row="31" column="1">
            <property name="text">
             <string>31</string>
            </property>
           </item>
           <item row="31" column="2">
            <property name="text">
             <string>0x00000000</string>
            </property>
           </item>
           <item row="32" column="0">
            <property name="text">
             <string>PC</string>
            </property>
           </item>
           <item row="32" column="1">
            <property name="text">
             <string/>
            </property>
           </item>
           <item row="32" column="2">
            <property name="text">
             <string>0x00400000</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Data Memory</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QTableWidget" name="tableDataMemory">
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>100</number>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Address</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>978</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_Instructions"/>
    <addaction name="actionLoad_Data"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuSimulation_Control">
    <property name="title">
     <string>Simulation Control</string>
    </property>
    <addaction name="actionExecute_All"/>
    <addaction name="actionNext_step"/>
    <addaction name="actionPrevious_Step"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSimulation_Control"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad_Instructions"/>
   <addaction name="actionLoad_Data"/>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionExecute_All"/>
   <addaction name="actionPrevious_Step"/>
   <addaction name="actionNext_step"/>
  </widget>
  <action name="actionLoad_Instructions">
   <property name="icon">
    <iconset resource="media.qrc">
     <normaloff>:/icons/binary-code.png</normaloff>:/icons/binary-code.png</iconset>
   </property>
   <property name="text">
    <string>Load Instructions</string>
   </property>
   <property name="statusTip">
    <string>Load a file with instructions in hexadecimal to instruction memory</string>
   </property>
  </action>
  <action name="actionLoad_Data">
   <property name="icon">
    <iconset resource="media.qrc">
     <normaloff>:/icons/hexsudoku.png</normaloff>:/icons/hexsudoku.png</iconset>
   </property>
   <property name="text">
    <string>Load Data</string>
   </property>
   <property name="statusTip">
    <string>Load a file with data to data memory</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionExecute_All">
   <property name="icon">
    <iconset resource="media.qrc">
     <normaloff>:/icons/playall.png</normaloff>:/icons/playall.png</iconset>
   </property>
   <property name="text">
    <string>Execute All</string>
   </property>
   <property name="statusTip">
    <string>Execute All Instructions</string>
   </property>
  </action>
  <action name="actionNext_step">
   <property name="icon">
    <iconset resource="media.qrc">
     <normaloff>:/icons/next.png</normaloff>:/icons/next.png</iconset>
   </property>
   <property name="text">
    <string>Next Step</string>
   </property>
   <property name="statusTip">
    <string>Go to next instruction</string>
   </property>
  </action>
  <action name="actionPrevious_Step">
   <property name="icon">
    <iconset resource="media.qrc">
     <normaloff>:/icons/previous.png</normaloff>:/icons/previous.png</iconset>
   </property>
   <property name="text">
    <string>Previous Step</string>
   </property>
   <property name="statusTip">
    <string>Go to previous instruction</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="media.qrc"/>
 </resources>
 <connections/>
</ui>
